### ISSUE TITLE
Interactive Transaction API timing out after upgrading from 4.7.0 and above

### ISSUE THREAD

**Issue by huboh** [prisma/prisma#18393]
After upgrading from Prisma version 4.4.0 to version 4.9.0, the Interactive Transaction API consistently times out for the same code that previously worked without issue. This issue appears to be specific to versions 4.7.0 and above, as this is when the Interactive Transaction API was introduced and normalized to the Prisma client without the need for the preview feature. It is unclear what may be causing the timeout, and further investigation may be required to determine the root cause.

## Client Snippet
```ts
// previous code which works
const someOp = data.classFeedbacks.map((cf) => prisma.classFeedBack.upsert({ where: { id: cf.id }, create: cf, update: {} }));
const anothersOp = data.campusTuitions.map((ct) => prisma.campusTuition.upsert({ where: { id: ct.id }, create: ct, update: {} }));

await prisma.$transaction([
    ...someOp,
    ...another
]);

// current code fails after update
await prisma.$transaction(
  async (txClient) => {
    .....
    await Promise.all([
      ...data.classFeedbacks.map((cf) => txClient.classFeedBack.upsert({ where: { name: cf.id }, create: cf, update: {} })),
      ...data.campusTuitions.map((ct) => txClient.campusTuition.upsert({ where: { name: ct.id }, create: ct, update: {} }))
    ])
  }
)
```

## Errror
```bash
thread 'tokio-runtime-worker' panicked at 'Internal error: Attempted to serialize empty result.', query-engine\core\src\response_ir\ir_serializer.rs:54:40
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

## Stack Trace 
```bash
Transaction API error: Transaction already closed: Could not perform operation.
    at Zr.handleRequestError (----project-path---\node_modules\@prisma\client\runtime\library.js:171:6414)
    at Zr.handleAndLogRequestError (----project-path---\node_modules\@prisma\client\runtime\library.js:171:5948)
    at Zr.request (----project-path---\node_modules\@prisma\client\runtime\library.js:171:5786)
    at async t._request (----project-path---\node_modules\@prisma\client\runtime\library.js:174:10455)
    at async Promise.all (index 0)
    at async prisma.$transaction.maxWait (----project-path---\scripts\seeder.ts:1162:21)
    at async Proxy._transactionWithCallback (----project-path---\node_modules\@prisma\client\runtime\library.js:174:9384) {
  code: 'P2028',
  clientVersion: '4.11.0',
  meta: { error: 'Transaction already closed: Could not perform operation.' }
}
```

**Comment by robertf224** [comment]
hey!  we actually thought this was interactive transactions as well at first, but we pinned it down to upserts, issue we filed a bit ago here https://github.com/prisma/prisma/issues/18285

@huboh you didn't mention but assuming you're also on MySQL

**Comment by huboh** [comment]
> Can you please make clearer what code exactly works in which version and which does not? Thanks.

Hey @janpio, I left comments in the `client snippet` section that explains which code works

**Comment by huboh** [comment]
@robertf224 I actually suspected multiple relations on a particular model was causing the code to fail. I was able to single out the model causing the error, it had 3 relations and we just had to find a way around that because i still haven't found a solution.

we're using mongo for the db.

```
...
model Payment {
  application_fee     Application?  @relation(name: "feePayment")
  application_initial Application?  @relation(name: "initialPayment")
  application         Application?  @relation(name: "applicationPayments", fields: [applicationId], references: [id])
  ...
```

**Comment by huboh** [comment]
@robertf224 I actually did suspected multiple relations on a particular model was causing the code to fail. I was able to single out the model causing the error, it had 3 relations and we just had to find a way around that because i still haven't found a solution.

we're using mongo for the db.

```prisma
model ModelName {
  ...
  application_fee     Application?  @relation(name: "feePayment")
  application_initial Application?  @relation(name: "initialPayment")
  application         Application?  @relation(name: "applicationPayments", fields: [applicationId], references: [id])
  ...
}
```

**Comment by MrDockal** [comment]
I faced a similar problem running many queries from my local machine against the remote DB. Increasing the transaction timeout fixed the problem for me https://www.prisma.io/docs/concepts/components/prisma-client/transactions#:~:text=timeout%3A%20The%20maximum%20amount%20of%20time%20the%20interactive%20transaction%20can%20run%20before%20being%20canceled%20and%20rolled%20back.%20The%20default%20value%20is%205%20seconds.

**Comment by janpio** [comment]
Are you using `relationMode=prisma` @huboh and @MrDockal?

**Comment by laplab** [comment]
This issue was fixed in [Prisma 5.13.0](https://github.com/prisma/prisma/releases/tag/5.13.0).
